### PR TITLE
Add interfaces for components

### PR DESCRIPTION
**Pull request by PrasadG193**

### CODE REVIEW

**Comment by sahil-lakhwani** [comment]
We'll have to modify the interface method name in interface or in the implementation

https://github.com/infracloudio/msbotbuilder-go/blob/04cf254af031e89b6103c2caa1cef1c32f33d70b/connector/auth/jwt_token_validation.go#L17-L26

File: core/bot_framework_adapter.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	if err != nil {
+		return activity, errors.Wrap(err, "Error while parsing Bot request")
+	}
+	return activity, bf.authenticateRequest(ctx, activity, authHeader)
+}
+
+func (bf *BotFrameworkAdapter) authenticateRequest(ctx context.Context, req schema.Activity, headers string) error {
+	jwtValidation := &auth.JwtTokenValidation{}
+
+	_, err := jwtValidation.AuthenticateRequest(ctx, req, headers, bf.CredentialProvider, bf.ChannelService)
```
Comment (sahil-lakhwani) [inline core/bot_framework_adapter.go]: There exists an interface in `jwt_token_validation.go` Should that be used? (After a few modifications) or this is okay?

Comment (PrasadG193) [reply core/bot_framework_adapter.go]: Aah, I missed that. Yeah, we should use that.

Comment (PrasadG193) [reply core/bot_framework_adapter.go]: Fixed! @sahil-lakhwani please rebase your PR. I am merging this

**Pull request merged**